staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -276,10 +277,17 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	if len(chain[0].Subject.CommonName) == 0 {
 		return nil, false, nil
 	}
+
+	fp := sha256.Sum256(chain[0].Raw)
+	id := "X509Fingerprint=" + hex.EncodeToString(fp[:])
```

Comment:
This prefix and value are effectively APIs, so worth thinking about the exact values we're producing.

For the name:
* Do we want to include the hash algorithm (SHA256) in the name or not? I lean towards being explicit, since fingerprints can also be computed with sha1.
* Do we want mixed case (e.g. `X509SHA256Fingerprint`) or separated upper-case (e.g. `X509_SHA256_FINGERPRINT`)?
* Do we need more than just `X509SHA256` as the credential ID? I could see this either way.
  * If so, are we happy with "fingerprint" terminology in the prefix name? That's the flag openssl uses and it's a term of art in RFCs, so it seems ok to me.

For the value:
* Does hex.EncodeToString produce upper or lower-case letters? Is that guaranteed to remain stable?
* Do we care which case we end up with? Looks like `openssl x509 ... -fingerprint -sha256` produces uppercase and `sha256sum` produces lowercase. Is there a fingerprint RFC that leans one way or the other?